pcap-haiku.c

```diff
@@ -50,20 +50,23 @@ static int
 pcap_read_haiku(pcap_t* handle, int maxPackets _U_, pcap_handler callback,
 	u_char* userdata)
 {
+	int32_t captureLength;
```

Comment:
Do these need to be moved?  We require C99 support in the main and 1.10 branches.

Reply:
You're right, these don't need to be moved. First I was testing with an older compiler and this was left in here like this.

Reply:
> I was testing with an older compiler

Using the configure script or CMake?  Both of them should require C99 support, but perhaps they're not testing for declarations in the middle of code.  What compiler was this?

Reply:
If you want to target GCC 2.95 you'll have to do C89 sadly, as it doesn't know about C99. Thinking of it, it might actually be the reason for the file being C++ in the first place, it's a cheap way of allowing C99-type features like variable declarations anywhere.

Reply:
The latest configure script only warns about the lack of C99 features, because, when generated by the 2.71 version of autoconf, it rejects the Solaris Studio 12.3 compiler without bothering to try -xc99, but the compiler handles libpcap anyway.  Maybe that's what happened.

So your choices are 1) leave the file in C++ or 2) require a newer compiler.  (Does Haiku really provide the 2.95 C compiler by default?  If so, does it also provide the 2.95 C++ compiler?)

Reply:
> If you want to target GCC 2.95 you'll have to do C89 sadly, as it doesn't know about C99.

Release History: GCC 2.95.3 March 16, 2001 !

Reply:
Not sure we want to support this rather old version.

Reply:
"It is not old, it is a classic!".

On a more serious note, GCC 2.95 has been the compiler of the 32-bit version of Haiku for a while (Haiku developers probably have a reason for that).

To me it would make a lot of sense to remove C++ from the problem space of libpcap so workarounds like those made in commit 914e40f (thank you Guy) would not be required in the solution space, and C89 does not seem an inconvenience.

Reply:
Yes we do have a reason: we want to be binary compatible with BeOS, which was written in C++, using GCC2, but GCC3 changed their C++ ABI to the Itanium one, and so we are stuck to 2.95 for the system libs.
Now, I'm not sure we want to keep supporting the GCC2 build, unless there's a system lib that depends on it, but [the changes seem to be minimal](https://github.com/haikuports/haikuports/blob/master/net-libs/libpcap/patches/libpcap-1.10.3.gcc2.patchset).

Reply:
I think it's only the 32-bit gcc2 build of tcpdump and nmap that relies on the gcc2 build of libpcap. We could migrate them to the gcc 11.2 toolchain and then we can do away with libpcap legacy build. (both libpcap and tcpdump needs a gcc2 patch to get a successful build, mostly variable declarations in multiple files)

Reply:
Probably we don't need to maintain gcc2 legacy toolchain compatibility for libpcap on Haiku as it's possible to migrate dependencies to more up-to-date gcc11 toolchain. On the other hand it wouldbe good to move the Haiku backend to C instead of C++.
Are there any outstanding review comments on this PR that I should follow up on?

Reply:
The C language's ABI isn't as complicated as the C++ language's ABI, so doing the 32-bit build of libpcap with a non-antique C compiler might not break anything.